plugins/com.google.cloud.tools.eclipse.appengine.ui/src/com/google/cloud/tools/eclipse/appengine/ui/AppEngineLibrariesSelectorGroup.java

```diff
@@ -104,87 +136,60 @@ private static String getLibraryName(Library library) {
-    Library library = (Library) libraryButton.getData();
-    ISWTObservableValue libraryButtonSelection =
-        WidgetProperties.selection().observe(libraryButton);
-    ISWTObservableValue libraryButtonEnablement = 
-        WidgetProperties.enabled().observe(libraryButton);
-    // library selection UI -> model
-    bindingContext.bindValue(libraryButtonSelection, new NullComputedValue(getDisplayRealm()),
-        new UpdateValueStrategy()
-            .setConverter(new HandleLibrarySelectionConverter(selectedLibraries, library)),
-        new UpdateValueStrategy(UpdateValueStrategy.POLICY_NEVER));
-    // UI <- library selection model
-    boolean resultIfFound = true;
-    bindingContext.bindValue(libraryButtonSelection,
-        new LibrarySelected(getDisplayRealm(), library.getId(), resultIfFound, libraryButton), 
-        new UpdateValueStrategy(UpdateValueStrategy.POLICY_NEVER), new UpdateValueStrategy());
-    // UI enablement <- library is a dependency
-    bindingContext.bindValue(libraryButtonEnablement,
-        new DependentLibrarySelected(getDisplayRealm(), library.getId(), false /* resultIfFound*/),
-        new UpdateValueStrategy(UpdateValueStrategy.POLICY_NEVER), new UpdateValueStrategy());
+  @VisibleForTesting
+  List<Button> getLibraryButtons() {
+    return new ArrayList<>(libraryButtons.values());
   }
 
-  public void dispose() {
-    if (bindingContext != null) {
-      bindingContext.dispose();
-    }
+  public void addSelectionChangedListener(ISelectionChangedListener listener) {
+    listeners.add(listener);
   }
 
-  private Realm getDisplayRealm() {
-    return DisplayRealm.getRealm(parentContainer.getDisplay());
+  public void removeSelectionChangedListener(ISelectionChangedListener listener) {
+    listeners.remove(listener);
   }
 
-  @VisibleForTesting
-  List<Button> getLibraryButtons() {
-    return libraryButtons;
+  /**
+   * Return the list of selected libraries and their required libraries.
+   */
+  public ISelection getSelection() {
+    return new StructuredSelection(getSelectedLibraries());
   }
 
-  private final class LibrarySelected extends DependentLibrarySelected {
-    private final Button libraryButton;
-
-    private LibrarySelected(Realm realm, String libraryId, boolean resultIfFound,
-        Button libraryButton) {
-      super(realm, libraryId, resultIfFound);
-      this.libraryButton = libraryButton;
+  /**
+   * Set selection by providing a collection of {@linkplain Library} objects or
+   * {@link Library#getId() library IDs}. All libraries are marked as being explicitly selected.
+   * Must be called from the SWT thread.
+   */
+  public void setSelection(ISelection selection) {
+    explicitSelectedLibraries.clear();
+    if (selection instanceof IStructuredSelection) {
+      for (Object object : ((IStructuredSelection) selection).toArray()) {
+        if (object instanceof String && availableLibraries.containsKey(object)) {
```

Comment:
`setSelection()` may be called with any selection.  Here we need to be very cautious.